core/src/openid.cpp

```diff
@@ -567,6 +576,30 @@ namespace irods::http::openid
 				return std::nullopt;
 			}
 
+			// In validating OIDC ID Tokens, if a symmetric algorithm is used,
+			// the private key of the algorithm is not defined if the token has
+			// multiple 'aud'.
+			// See OIDC Section 3.1.3.7
+			if (_type == TokenType::ID && alg.substr(0, 2) == "HS") {
+				// Assert there is only one 'aud' member
+				if (_jwt.get_audience().size() != 1) {
+					logging::error("{}: ID Token with symmetric [alg] and multiple [aud] is not supported.", __func__);
+					return std::nullopt;
+				}
+			}
+
+			// In validating OIDC ID Tokens, if the 'azp' claim is present,
+			// we may verify that our 'client_id' is in the claim.
+			// See OIDC Section 3.1.3.7
+			if (_type == TokenType::ID && _jwt.has_payload_claim("azp")) {
+				if (_jwt.get_payload_claim("azp").as_string() !=
+				    irods::http::globals::oidc_configuration().at("client_id").get_ref<const std::string&>())
+				{
+					logging::error("{}: ID Token [azp] of type [none] is not supported.", __func__);
```

Comment:
Is `type [none]` in this log message correct?

The conditionals leading up to it imply it will hold a string of some value.

Reply:
Fixed.